homeassistant/helpers/trigger_template_entity.py

```diff
@@ -389,3 +391,28 @@ def __init__(
         ManualTriggerEntity.__init__(self, hass, config)
         self._attr_native_unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
         self._attr_state_class = config.get(CONF_STATE_CLASS)
+
+    @callback
+    def _set_value_with_possible_timestamp(
```

Comment:
I'm not sold on this method also calling `_process_manual_data`, it doesn't match the method name and has nothing to do with the special handling of timestamps.